{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4066c751",
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "import pandas as pd\n",
    "\n",
    "# Set GitHub Personal Access Token\n",
    "ACCESS_TOKEN = ''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "564088e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "## Getting paginated repos:\n",
    "## If the project has only one repo, you do not need this segment\n",
    "\n",
    "import requests\n",
    "\n",
    "# Replace 'owner_username' with the GitHub username of the owner\n",
    "owner_username = ''\n",
    "\n",
    "# GitHub API endpoint for repositories of a specific user\n",
    "api_url = f'https://api.github.com/users/{owner_username}/repos'\n",
    "\n",
    "try:\n",
    "    # List to store all repository names\n",
    "    all_repository_names = []\n",
    "    \n",
    "    # Variable to store the page number\n",
    "    page = 1\n",
    "    \n",
    "    while True:\n",
    "        # Make a GET request to the GitHub API for the current page\n",
    "        response = requests.get(api_url, params={'page': page, 'per_page': 100})  # Fetch 100 repositories per page\n",
    "        \n",
    "        # Check if the request was successful (HTTP status code 200)\n",
    "        if response.status_code == 200:\n",
    "            # Parse the JSON response\n",
    "            repositories = response.json()\n",
    "            \n",
    "            # If there are no more repositories on the current page, break the loop\n",
    "            if len(repositories) == 0:\n",
    "                break\n",
    "            \n",
    "            # Extract repository names from the current page and add to the list\n",
    "            repository_names = [repo['name'] for repo in repositories]\n",
    "            all_repository_names.extend(repository_names)\n",
    "            \n",
    "            # Move to the next page\n",
    "            page += 1\n",
    "        else:\n",
    "            # Print an error message if the request was not successful\n",
    "            print(f'Error: Unable to fetch repositories. Status code: {response.status_code}')\n",
    "            break\n",
    "    \n",
    "    # Print all repository names\n",
    "    print(f'Repositories of {owner_username}: {all_repository_names}')\n",
    "except Exception as e:\n",
    "    # Handle exceptions\n",
    "    print(f'An error occurred: {e}')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a8cb5523",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GitHub issues data for galaxyproject (open and closed) saved to github_issues.csv\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# Define Function to Get All Issues for a Repository (Open and Closed)\n",
    "def get_all_repository_issues(owner, repo):\n",
    "    url = f'https://api.github.com/repos/{owner}/{repo}/issues?state=all'\n",
    "    headers = {'Authorization': f'token {ACCESS_TOKEN}'}\n",
    "    all_issues = []\n",
    "    page = 1\n",
    "    while True:\n",
    "        params = {'page': page, 'per_page': 30000}  # 30000 issues per page, adjust per your needs\n",
    "        response = requests.get(url, headers=headers, params=params)\n",
    "        if response.status_code == 200:\n",
    "            issues = response.json()\n",
    "            if len(issues) == 0:\n",
    "                break\n",
    "            all_issues.extend(issues)\n",
    "            page += 1\n",
    "        else:\n",
    "            print(f'Error fetching issues from {owner}/{repo}. Status code: {response.status_code}')\n",
    "            break\n",
    "    return all_issues\n",
    "\n",
    "# Get Repositories for a Specific Owner\n",
    "owner = 'galaxyproject'  # Replace 'OWNER_NAME' with the GitHub username of the owner\n",
    "\n",
    "repositories = []\n",
    "url = f'https://api.github.com/users/{owner}/repos'\n",
    "headers = {'Authorization': f'token {ACCESS_TOKEN}'}\n",
    "response = requests.get(url, headers=headers)\n",
    "if response.status_code == 200:\n",
    "    repositories = [repo['name'] for repo in response.json()]\n",
    "else:\n",
    "    print(f'Error fetching repositories for {owner}. Status code: {response.status_code}')\n",
    "\n",
    "# Get All Issues (Open and Closed) from Repositories\n",
    "all_issues = []\n",
    "## Put your Obtained Repositoires in here\n",
    "#repositories = ['2015_Bray_GigaScience-data', '2016_10_19_InSitu_Expression_Hacohen_Lab_Paul_Hoover_Partners', '2017_05_10_neutrophile_inhibitor', '2019_scWorkshop', '2020-06-01-Evidence-of-state-switching-in-single-cell-drug-response-Broad', '2020-goodman-lee', '2020_04_07_COVID_Davey_lab', '2020_scWorkshop', '2021_04_07_ProteinKinaseInhibitors_NathanaelGray', '2021_09_01_VarChAMP', '2021_09_01_VarChAMP-data', '2022_09_07_New_phenotypic_dye_testing_CDoT_Broad', '2022_PERISCOPE', '2023_cytominer_sphering_benchmark', '3dmutvis', 'ABC-Enhancer-Gene-Prediction', 'accelerate_simple_query', 'achilles_onboarding', 'achilles_sheet_scraper', 'action-shellcheck', 'action-slack', 'action-yamllint', 'adam', 'adapt', 'adapt-analysis', 'adapt-designs', 'adapt-designs-continuous', 'adapt-pipes', 'adapt-seq-design', 'adapt-web', 'agora', 'akka-statsd', 'alfre', 'alicia', 'AMARETTO-Hub', 'AmpSeQC', 'analytics-public', 'AncestralGeneRator', 'Aneuploidy_dependencies', 'anorm221', 'ansible-role-basepackages', 'ansible-role-configure', 'ansible-role-consul', 'ansible-role-docker', 'ansible-role-docker-overlayfs', 'ansible-role-docker-service', 'ansible-role-docker-thinpool', 'ansible-role-filebeat', 'ansible-role-jenkins', 'ansible-role-jenkins-user', 'ansible-role-local-filesystem', 'ansible-role-lvm', 'ansible-role-os-settings', 'ansible-role-rhel-hardening', 'ansible-role-sbt', 'ansible-role-sentinelone', 'ansible-role-stackdriver', 'ansible-role-token-copy', 'ansible-role-yum-cron', 'ansible-yum-cron', 'antenna', 'antryg', 'anvil_tdr_ingest', 'aou-api-prototype-java7', 'aou-researcher-webui-react-prototype', 'aou-ui-angular-prototype', 'apache-ubi-proxy-fips', 'apache2-onehealth', 'APSGhistory', 'arbio', 'argo-helm', 'atestrepo', 'aurora', 'auspice', 'AuSPICES', 'author-affiliations', 'automerge-action', 'automop', 'AutoTrain', 'Avant-garde-Terra', 'avro-import', 'AwesomeGenomics', 'aws-backend', 'aws-cdk-research-computing', 'aws-cloudformation-templates', 'axios-retry', 'Azure-Sentinel', 'azure-warp-joint-calling', 'BactRNASeqCount', 'bam_sampler', 'barclay', 'BARD', 'barkbase_paper_analysis', 'baseimage', 'batch-effect-correction', 'BBBC021_Morpheus_Exercise', 'bcarmen', 'bdg-formats', 'bdg-services', 'be-validation-pipeline', 'beacon-of-beacons', 'beast-beagle-cuda', 'beast-beagle-pocl', 'beast2-beagle-cuda', 'beehive', 'bibt-gcp-asset', 'bibt-gcp-scc', 'bibtutils', 'BioC2020Workshop_AMARETTO-Hub', 'biomarker_lite', 'bioscala', 'bioshop', 'BioSimSpaceTutorials', 'biowdl-test-data', 'BipolarCell2016', 'bits-aoc-leaderboard', 'bits-ap-modules', 'bits-appengine', 'bits-backupify', 'bits-chef-repo', 'bits-docker', 'bits-fedex', 'bits-gpgfile', 'bits-progressbar', 'bits-terraform-modules', 'bitstore-app', 'bmxp', 'bQTL_mapping_rerun', 'bri', 'broad-babel', 'broad-epi-cut-n-run', 'broad-epi-repeats-analysis', 'broad-epi-workflows-collection', 'broad-fungalgroup', 'broad-translator', 'broadinstitute.github.io', 'brown_lab_warp', 'BTL-EntryCreator', 'BTL-MdReport', 'BTL-Parsomatic', 'BTL-tracker', 'btl_firecloud', 'btl_firecloud_gatk', 'busyflow.pivotal', 'bvas', 'bz_bbf', 'cakephp', 'cakephp-mongodb', 'capirca-syntax-highlighter', 'carmen-rvp', 'carrot', 'carrot-example-test', 'carrot-publish-github-action', 'carrot_cli', 'castnet', 'catch', 'cats-effect', 'CCLE_methylation', 'CCLF_TWIST', 'CDoT_compound_track_2', 'cdot_track_compounds_run', 'cds-ensemble', 'cdsr_biomarker', 'cdsr_gsea', 'cdsr_models', 'cdsr_plots', 'CEGS-2018', 'CEGS_single_cell_tx_workshop_Sept2017', 'cell', 'cell-health', 'cell-painting-batch-effects', 'cell-painting-vae', 'cellarium', 'CellBender', 'celligner', 'celligner2', 'Celligner_ms', 'CellMincer', 'CellMiner', 'cellpainting-gallery', 'cellpainting-gallery-validator', 'cellpainting_scripts', 'cellprofiler-on-Terra', 'cellprofiler-terra', 'celseq2-wdl', 'centaur', 'ces-incipient', 'cfn-pyplates', 'chart-releaser-action', 'chart-testing-action', 'charts', 'chbv', 'checklists.github.io', 'checkpoint-on-preemption', 'chef-docker', 'chef-shibboleth_idp', 'CHIP_EWAS', 'chowser', 'chronos', 'circleci-orb-bits-puppet', 'circleci-orb-bits-python', 'CisRegModels', 'CLASTR-internal', 'clio', 'cloud-cost-management', 'cma-search', 'cmi-cas-rest-client', 'cmQTL', 'cms', 'cms2', 'cms2-docker-component-stats', 'cms2-pipelines', 'cms2-work', 'CNVsingle', 'cn_correlation', 'codecov-bash-uploader', 'CODECsuite', 'coessentiality', 'CommunityAMARETTO', 'comodo_rest_api', 'COMOKIT-Model', 'ComparatorByTrait', 'comparing-moa-representations', 'computational-resources', 'Concensus2', 'concensusGLM', 'consensus_svs', 'conseq', 'ConSequences', 'ConsolePi', 'cookiecutter-bits-pypi', 'cosi2', 'cotools', 'cotton', 'covid-19-static', 'covid19-altruistic-testing', 'covid19-sequencing', 'covid19-testing', 'covisr', 'CP186-A549-WG', 'CP257-HeLa-WG', 'cpg0017-rohban-pathways', 'cptac-convert-to-gct', 'create-pull-request', 'CRISPRi-pipeline', 'CRISPRiTilingDesign', 'CRM2.0', 'CromIAM', 'cromshell', 'cromshell-1', 'cromstream', 'cromwell', 'cromwell-accountant', 'cromwell-backend', 'cromwell-client', 'cromwell-develop', 'cromwell-monitor-deprecated', 'cromwell-task-monitor-bq', 'cromwell-tools', 'cromwhelm', 'crowd-jaas', 'CTAT-pVACseq', 'CTAT-VirusIntegrationFinder', 'ctc_firehose', 'cumulonimbus', 'cwl-proto', 'cxowrapper', 'cytominer_scripts', 'dagster-utils', 'damage', 'data-provider-service', 'datacentred-ldap', 'datarepo-actions', 'datarepo-helm', 'datarepo-helm-definitions', 'datarepo-umbrella-release-action', 'dcp-community', 'ddp-android-sdk', 'ddp-angular', 'ddp-study-manager', 'ddp-study-manager-ui', 'ddp-study-server', 'ddp_monster_ingest', 'DecisionAnalysis', 'deepometry', 'DeepProfilerExperiments', 'del_sequence_processing', 'demuxEMS', 'denimcoat', 'deployment-status', 'deployment-template', 'depmap-api', 'depmap-crispr-vs-rnai', 'depmap-django-forestadmin', 'depmap_omics', 'depmap_omics_upload', 'depmap_target_birc6', 'dhcpd-cli', 'dig-aggregator-core', 'dig-aggregator-method.g8', 'dig-aggregator-methods', 'dig-aws', 'dig-bioindex', 'dig-diabetes-portal', 'dig-dug-portal', 'dig-dug-server', 'dig-dug-static-resources', 'dig-indexer', 'dig-kb-openapi-specs', 'dig-loam', 'dig-loam-images', 'dig-loam-stream', 'Dino4Cells_analysis', 'DINO4Cells_code', 'discovar_de_novo', 'disk-manager', 'DLBCL_Nat_Med_April_2018', 'docker-buckettest', 'docker-clojure-node', 'docker-cloudsqlproxy', 'docker-cr-rotate', 'docker-devnull-vscode-puppet', 'docker-devnull-vscode-python', 'docker-elasticsearch', 'docker-flyway', 'docker-gam', 'docker-gcds', 'docker-gradle', 'docker-hastebin', 'docker-liquibase', 'docker-mongo', 'docker-nedi', 'docker-nginx', 'docker-nmap', 'docker-orientdb', 'docker-pyplates', 'docker-scala', 'docker-sentry-github', 'docker-slapd', 'docker-stash', 'docker-terraform', 'docker-timefix', 'docker-vault-server', 'docker-vaultcli', 'dockerized_mts', 'dockstore-tool-bamstats', 'dockstore-tool-cms2', 'dockstore-tool-cosi2', 'dockstore-tool-selscan', 'dos-drs-perf', 'dpg_static_content', 'Drop-seq', 'DropletUtils', 'DropSeq_utility', 'dropviz', 'dsde-deep-learning', 'dsde-probabilistic-programming', 'dsp-ansible-configs', 'dsp-appsec-actions-docker-gcr', 'dsp-appsec-actions-gke-poc', 'dsp-appsec-base-image-hardening', 'dsp-appsec-centos7-hardening-role', 'dsp-appsec-codedx-api-client-python', 'dsp-appsec-debian9-hardening-role', 'dsp-appsec-infrastructure-apps', 'dsp-appsec-sast', 'dsp-appsec-seckb', 'dsp-appsec-sphinx-kb-archived', 'dsp-appsec-terra-compliance-poc', 'dsp-appsec-testing', 'dsp-appsec-trivy-action', 'dsp-appsec-trivy-cicd', 'dsp-appsec-zap-automation', 'dsp-backstage', 'dsp-data-models', 'dsp-devops-wiki', 'dsp-grafana', 'dsp-ingest-deploy', 'dsp-reusable-workflows', 'dsp-terra-mooc-intro', 'dspman', 'DSP_User_Ed', 'Dymecki-neurons', 'EasyTipView', 'EbolaNaturalHistory', 'elba', 'emerald-kubernetes-kafka', 'encode_utils', 'ensemble_predictor', 'epi-correlation', 'epi-diffbind', 'epi-lims-workflows', 'epi-ms-shareseq_viewer.github.io', 'epi-SHARE-seq-pipeline', 'epi-wdl-workshop', 'eqtl_pipeline_terra', 'eroc', 'eugene', 'evangalion', 'exome-results-browsers', 'exome_penetrance', 'ExP-model-fit', 'eyp-initscript', 'fastq_sampler', 'fcgdctools', 'FeatureAggregation', 'FeatureAggregation_single_cell', 'fiabftw', 'file-stats', 'filestats-bq', 'firecloud-bucket-log-shipper', 'firecloud-cli', 'firecloud-explorer', 'firecloud-orchestration', 'firecloud-tools', 'firecloud-ui', 'firecloud-ui-browser-tests', 'firecloud_developer_toolkit', 'firepony', 'fiss', 'FiTnEss', 'flask-restplus', 'flipbook', 'for_ucsc', 'fungal-wdl', 'FungalDx_CandidaID', 'funnel', 'funpipe', 'FusionInspectorPaper', 'g2p_molart', 'GAEMR', 'gamgee', 'Gap2Seq', 'garethr-docker', 'gatk', 'gatk-bwamem-jni', 'gatk-dataflow', 'gatk-docs', 'gatk-evaluation', 'gatk-fermilite-jni', 'GATK-for-Microbes', 'GATK-For-Primates', 'GATK-gCNV-publication', 'gatk-minimap2-jni', 'gatk-native-bindings', 'gatk-protected', 'gatk-sv', 'gatk-sv-1', 'gatk-sv-2', 'gatk-sv-3', 'gatk-sv-4', 'gatk-sv-genotyper', 'gatk-sv-single-sample', 'gatk-tool-wdls', 'gatk3-4-rnaseq-germline-snps-indels', 'gatk4-data-processing', 'gcp-dm-templates', 'gcs-list-md5', 'GDAN', 'gdata-cli', 'gdctools', 'gdr-ingest', 'GeLiNEA', 'gene-hints', 'gene-mashup', 'genepy', 'GeneSelection', 'genesets', 'genetics-kp-creative', 'genetics-kp-dev', 'gene_knockout_correlation', 'gene_select_v3', 'genomespace_dot_org', 'genomestrip-builds', 'genomestrip-cnv-browser', 'genomestrip-gatk', 'genomestrip-terra', 'genomics-in-the-cloud', 'genus', 'getzlab-LNP', 'getzlab-LNPHS_utils', 'getzlab-PCAWG-MutSig2CV_NC', 'getzlab-PCAWG-power_calculations', 'getzlab-PCAWG-pvalue_combination', 'getzlab-PHS', 'gh-action-pypi-publish', 'gisaid-cli', 'gistic2', 'git-secrets', 'GitDescribe-Maven-Plugin', 'github-action-consul-template', 'github-action-get-previous-tag', 'github-action-push', 'github-action-template-render', 'github-app-creds-example', 'github-maven-package-example', 'github-tag-action', 'github-vault-action-usage', 'glmm-de-genes', 'gnomad-blog', 'gnomad-browser', 'gnomad-browser-toolkit', 'gnomad-constraint', 'gnomad-helm', 'gnomad-mitochondria', 'gnomad-project-template', 'gnomad-readviz', 'gnomad_bigquery', 'gnomad_chets', 'gnomad_local_ancestry', 'gnomad_lof', 'gnomad_methods', 'gnomad_qc', 'gnomad_review_hum_mut', 'godror', 'golang-project-template', 'Google-Apps-Script', 'google-cloud-auto-snapshot', 'google-cloud-java', 'google-storage-ftp', 'GoogleBillingDashboard', 'gotc-helm-repo', 'gp-crspjanusapp', 'gp-oxfordsync', 'gpalloc', 'GPP-bulk-rnaseq-protocols', 'gpp-rnaseq-analysis', 'GpUnit', 'gradle-command-action', 'grafana-variable-aliaser', 'graphite-experiment', 'green', 'green-helm-repo', 'grit-benchmark', 'gsalib', 'gtex-ashg2017-workshop', 'gtex-pipeline', 'gtex-public', 'gtex-single-nucleus-reference', 'gtex-tutorials', 'gtex-v8', 'gtex-viz', 'gumbo_client', 'gwas-ui', 'HandLens', 'HandLens-1.0.0', 'hca-adapter-tools', 'hdf5-java-bindings', 'helm-argo-test', 'helm-fluentd', 'helm-scala-sdk', 'helm-values-example', 'herc', 'hermitcrab', 'heterodon', 'hiring-without-whiteboards', 'hjf-playground', 'hmp2-ibdmdb-cms', 'hmp2-ibdmdb-results', 'homebrew-core', 'homebrew-dsp', 'honeypot', 'horsefish', 'hostinfo', 'hscgp', 'HTAPP-methods-supp', 'htslib', 'hybsel-design-runs', 'HydrantFC', 'iam_kernel_poc', 'iam_poc_ui_client', 'ichorCNA', 'identify', 'ID_Sandbox', 'igv-experiments', 'igv.js', 'image-profiling-workflow-template', 'imaging-backup-scripts', 'imaging-bbbc', 'imaging-platform-pipelines', 'imaging-platform-terraform', 'ImagingAMARETTO', 'imaging_metric_comparison', 'import-service', 'import_service_terraform', 'infercnv', 'infercnvApp', 'infercnvNGCHM', 'inferCNV_examples', 'infercnv_workshop', 'infrastructure-agent-puppet', 'inspec-gcp-cis-benchmark', 'inspec-gke-cis-benchmark', 'inspec-k8s', 'install-gcs-connector', 'IntelliSleepScorer', 'Intro_python_CPAcademy', 'ip-superurop-t_gao', 'isham_wgs', 'Italiano-MK-Analysis', 'jacalloc', 'jade-data-repo-client-example', 'jati', 'java', 'java-baseimage', 'jedi-public', 'jinja2-gcp-secret-manager', 'jSnap', 'juniper', 'junkbox', 'kaniko-cloudsdk-base-bug', 'kaniko-stage-caching-bug', 'kaniko-stage-naming-bug', 'kdux-rails-baseimage', 'keras-imaging', 'keras-rcnn', 'keras-resnet', 'KrumlovSingleCellTx2018', 'KrumlovSingleCellWorkshop2020', 'kryo', 'labs-on-cloud', 'lander-journalclub', 'language-wdl', 'lattice', 'lenthall', 'leonardo-cron-jobs', 'leonardo-libs', 'lgtm-pr-test-repo', 'liftover', 'lincs-cell-painting', 'lincs-profiling-complementarity', 'lloyd', 'Loggly-Puppet', 'lomax', 'long-read-pipelines', 'longbow', 'lr-automation', 'lr-malaria-automation', 'lrma-cloud-utils', 'luad-cell-painting', 'lunaris', 'lxc', 'MAESTRO-probe_designer', 'malaria-amplicon-pipeline', 'malaria-vc', 'MantaSVAnnotator', 'martha', 'MAS-ISO-seq', 'mas-seq-paper-data', 'mcarmen', 'mccarroll-cromwell', 'mcfeely', 'MD-IFP', 'MDL-workflows', 'mea-cas13', 'mea-cas13-analysis', 'megadrile', 'metabaron', 'MFmap', 'minerva_tma_codex', 'minnie-kenny', 'mira-SARS-CoV-2', 'mirrors-pylama', 'mistsystems', 'mix_seq_ms', 'ml4h', 'ml4ht_data_source', 'mocasa', 'mock-provider', 'molecular-data-provider', 'monster', 'monster-deploy', 'monster-etl', 'monster-helm', 'monster-ingester', 'monster-sbt-plugins', 'monster-scio-utils', 'monster-storage-libs', 'monster-xml-to-json-list', 'MorphEm', 'MPG-Hack_2', 'mutect', 'MuTest', 'mvicuna', 'mysql', 'ncbi-tools', 'ncbiaccess-test', 'NeMO-templates', 'netbox-docker', 'neural-profiling', 'neural-profiling-archive', 'nextstrain-builds', 'NGCHMR', 'nitrogenase', 'oglhclient', 'ohai', 'ols-docker', 'omim-search', 'oncotator', 'one-shot-atlas', 'ontology-service', 'OO-modeling', 'openidc-baseimage', 'openidc-proxy', 'openidc-terra-proxy', 'openldap', 'operation-outbreak-data-models', 'OpticalElectrophysiology', 'OptionsMenu', 'ORFCall', 'orsp-api', 'orsp-pub', 'ossec-puppet', 'pagerduty-trigger-slack-command', 'palantir-workflows', 'PangenomeVis', 'pangolin', 'PANOPLY', 'ParallelDonorAssignment', 'pasec', 'password-alert', 'PatHCapProbes', 'PatHCap_PL', 'PaulaLlanos_Projects', 'pav-wdl', 'pbs-analysis', 'pcawg_public', 'pcull', 'pdb2pqr', 'pdm-backend', 'pdm-ui', 'pe2loaddata', 'peat', 'PEHaplo', 'pepper', 'pepper-cmi-angular', 'pepper-cmi-backend', 'PerMutation', 'PerturbationAMARETTO', 'pgen-jni', 'phenet', 'phenics', 'phenotype-driven-analysis', 'phewas_genetic_analysis', 'phlock', 'PhylogicNDT', 'picard', 'pilon', 'pipelines-tools', 'PipelinesSupport', 'playwright-e2e', 'plink2-docker', 'plinkdocker', 'plotly-dash-gcp', 'podman', 'policyscanner', 'polyphonia', 'pooled-cell-painting-image-processing', 'pooled-cell-painting-profiling-recipe', 'pooled-cell-painting-profiling-template', 'poolq', 'popscle', 'position-effect-correction', 'potato-site', 'PowerShell-Docker', 'prancer', 'pre-commit-hooks', 'preemptible_cromwell', 'prism_data_processing', 'prism_reference_oncology', 'profiling-recipe', 'profiling-resistance-mechanisms', 'Protein_Kinase_Inhibitors_Gray', 'proteomics', 'proteomics-CPTAC-BKM120', 'proteomics-CPTAC-BRCA2016', 'proteomics-CPTAC-BRCA2020', 'proteomics-CPTAC-LSCC2021', 'proteomics-CPTAC-LUAD2020', 'proteomics-CPTAC-MONTE2022', 'proteomics-DIA-VMCollapsing', 'proteomics-PTM-DIA', 'proteomics-ptmsigdb-app', 'proteomics-SigPath-supplemental-data', 'protigy', 'PRSmix', 'ptm-normalization-script-test', 'PuckCaller', 'pupmod-simp-auditd', 'puppet-amavisd', 'puppet-autofs', 'puppet-avamar', 'puppet-ca_cert', 'puppet-certificates', 'puppet-certs', 'puppet-certs-1', 'puppet-clamav', 'puppet-cups', 'puppet-dell', 'puppet-dhcp', 'puppet-dns', 'puppet-dovecot', 'puppet-duo_openvpn', 'puppet-fail2ban', 'puppet-fluentd', 'puppet-foreman', 'puppet-foreman_proxy', 'puppet-freeradius', 'puppet-gcloudsdk', 'puppet-google_cloud_sdk', 'puppet-hiera', 'puppet-katello', 'puppet-logrotate', 'puppet-mlocate', 'puppet-module-cgroups', 'puppet-module-epel', 'puppet-module-nrpe', 'puppet-mongodb', 'puppet-nsswitch', 'puppet-openam', 'puppet-opendj', 'puppet-opendkim', 'puppet-openldap', 'puppet-openssl', 'puppet-openvpn', 'puppet-pdsh', 'puppet-php', 'puppet-portmap', 'puppet-postfix', 'puppet-qpage', 'puppet-qpid', 'puppet-qualys_agent', 'puppet-r', 'puppet-rssh', 'puppet-rsyslog', 'puppet-saslauthd', 'puppet-sentinelone_agent', 'puppet-slurm', 'puppet-snmp', 'puppet-splunk', 'puppet-ssh', 'puppet-stackdriver-agent', 'puppet-tomcat', 'puppet-tuned', 'puppet-ulimit', 'puppet-yp', 'puppet-zabbix', 'puppetlabs-certregen', 'puppetlabs-firewall', 'puppetlabs-puppetdb', 'puppetlabs-puppet_agent', 'push-to-cloud-service', 'py3-bio', 'pyfrost', 'PyGithub', 'pygls', 'pylama', 'pyqtl', 'pyro-cov', 'python-alteon', 'python-cert_manager', 'python-clearpass', 'python-mistery', 'python-slack-scim', 'python-standard-bits', 'python-sudoers', 'pytorch_fnet', 'pytos', 'pywdl', 'qbsub', 'qiime2', 'quickboard', 'raman_classifier_challenge', 'ras-passport-example', 'RASPDplus', 'rawls', 'rd-methods', 'react-broad', 'react-broad-bootstrap', 'react-cljs-modal', 'react-cljs-nav', 'reaper', 'reasoner', 'rec-seq', 'regevlab-jupyter-docker', 'regional_missense_constraint', 'repeatVariantsInCCLE', 'repository-dispatch', 'repurposing', 'ResearchKit', 'retry', 'revere', 'rigerj', 'rl_reasoner', 'RNA_MUTECT_1.0-1', 'rna_seq_sars_cov_2', 'rs3_manuscript', 'RVBLR', 'sabeti-dhis2-clinical', 'sabeti-dhis2-clinical-express', 'sabeti-host-txnomics-internal', 'sabeti-publichealthwriting', 'sabetilab-remote-config', 'sails-mongo', 'sam', 'samplyzer', 'sassy', 'SatMut_ASM_Parser', 'SatMut_VariantLibrary_Designer', 'sc2-cmu-study', 'sc2-rmd', 'sc2-seq-dash-public', 'sc2-variation-scripts', 'scala-baseimage', 'scalable_analytics', 'scalacache', 'scaraplate', 'scientific-python-2.7', 'scion', 'SCN2A_Base_Editing', 'scp-ingest-pipeline', 'scp-rails-baseimage', 'scred', 'scripts_notebooks_fossa', 'scRNA-Seq', 'sc_analysis_portal', 'sc_topics', 'security_monkey', 'segmentation', 'segmentation_experiments', 'selene', 'sensu-chef', 'sensu-community-plugins', 'seQc_MultiQC', 'seqr', 'seqr-helm', 'seqr-loading-pipelines', 'sequence-upload-to-gs', 'seQuoia', 'servicenow-powershell', 'setup-chart-releaser', 'setup-helmfile', 'setup-vault', 'shared-workflows', 'sharpener', 'sharpener-attribute-filter', 'sharpener-bicluster-expander', 'sharpener-common-to-rare-disease-gene-producer', 'sharpener-depmap-expander', 'sharpener-disease-genes-producer', 'sharpener-expression-correlation', 'sharpener-geneset-producer', 'sharpener-gui', 'sharpener-hypergeometric-enrichment-expander', 'sharpener-hypergeometric-enrichment-filter', 'sharpener-pathway-enrichment', 'sharpener-phenotype-similarity-expander', 'sharpener-random-producer', 'sharpener-remove-genes-filter', 'sharpener-string-expander', 'sharpener-tcrd-filter', 'sherlock', 'shibboleth-service-provider', 'SignatureAnalyzer-GPU', 'SimCLR3D', 'single-cell-annotation', 'single-cell-images', 'SingleCell_Morphological_Analysis', 'single_cell_analysis', 'single_cell_classification', 'single_cell_portal', 'single_cell_portal_core', 'skopy', 'slats', 'sldsc-eqtls', 'sl_joint_genotyping_test', 'sma-finder', 'smartseq2-single-nucleus-multisample', 'smilesDrawer', 'smn_analysis', 'smoofun', 'snakemake-broad-uger', 'snoop', 'SnowmanSV', 'SNPiR-Python', 'snpsnap', 'snputil', 'sntr', 'soerenbe-syslog_ng', 'somatic-benchmark', 'spames', 'sparklespray', 'speady', 'spiderdot', 'SpliceAI-lookup', 'SpliceAI-lookup-dev', 'split_merge_pl', 'spoa-rs', 'SPR_Create_Dotmatics_ADLP_File', 'Squeryl', 'SRA-submission', 'sra_submission_tool', 'ssGSEA2.0', 'ssl-cert-test', 'starfire', 'stimATAC_analyses_code', 'str-analysis', 'str-truth-set', 'StrainGE', 'strainge-paper', 'StrPileups', 'sttp-swagger-codegen', 'stxxl', 'submissions', 'subsampler', 'SVGKit', 'sw-test-distroless', 'swagger-ui-xss-poc-fix', 'SynerClust', 'sz-gatk-sv-vjalili', 't3-support-script', 'TAG-public', 'taiga', 'taigapy', 'taigr', 'tangent', 'tango', 'Tangram', 'tcrd-api', 'TCRP', 'temporal-rna-seq-deconvolution', 'tensorqtl', 'tentacle', 'terra-featured-workspace-tests', 'terra-fw-notebooks', 'terra-helm', 'terra-helm-definitions', 'terra-helmfile-images', 'terra-import-builder', 'terra-incident-response', 'terra-jupyter-plotly', 'terra-profile', 'terra-terms-of-service', 'terra-tools', 'terraform', 'terraform-ap-modules', 'terraform-demo', 'terraform-github-actions', 'terraform-google-monitoring-uptimecheck', 'terraform-jade', 'terraform-kubernetes-google-secret-accessor', 'terraform-shared', 'terraform-terra', 'terranblib', 'test-jekyll-vdauwera', 'TestArsen2', 'test_rep', 'tgg-sredocs', 'tgg-terraform-modules', 'tgg-viewer', 'tgg_methods', 'theforeman-certs', 'theforeman-foreman', 'thelma', 'thrift', 'thurloe', 'TigerETL2', 'tim-retreat-breakout-session-2021', 'tinyquery', 'tissue_purifier', 'tmpnb', 'toil', 'TOPMed_AFib_pipeline', 'travis-reporter', 'trinityrnaseq', 'trivy-cve-scan', 'TSCA_firecloud_WDLS', 'ttn-neb-analysis', 'tubo', 'tups', 'turnstyle', 'udix', 'uger-cromwell', 'ukbb_qc', 'umi-collapser-internal', 'UMINormalize', 'update-gene-symbols-hgnc', 'uploader', 'v-phaser2', 'vagrant', 'vagrant-bases', 'vagrant-puppet', 'variant-curation-portal', 'VariantBam', 'varianteval', 'variantreviewparty', 'variants-cosmos-spikes', 'variantstore', 'various', 'vault', 'vault-api', 'vault-boss', 'vault-common', 'vault-crd-helm', 'vault-datamanagement', 'vault-init', 'vectors4s', 'villages-sqtl-pipeline', 'vim-wdl', 'viral-assemble', 'viral-baseimage', 'viral-classify', 'viral-core', 'viral-ngs', 'viral-ngs-deploy', 'viral-ngs-staging', 'viral-phylo', 'viral-pipelines', 'viral-rmd', 'viral-workshops', 'vlad', 'warp', 'warp-tools', 'wdl', 'wdl-for-dockstore', 'wdl-ide', 'wdl-pipseeker', 'wdl-runner', 'wdl-sublime-syntax-highlighter', 'wdl-syntax-highlighting', 'wdl4s', 'wdltool', 'wes2cromwell', 'wfl', 'widdler', 'winstanley', 'woodard', 'wordpress-crowd-plugin', 'workbench-libs', 'workbench-tos', 'workflow-dispatch', 'Workflow-Visualizer', 'workingdogs', 'wot', 'xtermcolor', 'yale', 'yardstick-app', 'yardstick-compound-structure', 'yootilz', 'Zoonomia']\n",
    "repositories = ['gops', 'gravity', 'gsoc', 'GTN-community-paper-2020', 'gx-it-proxy', 'gx-tool-db', 'gx-tool-db-panel-sync', 'gx-tool-db-project', 'gxabm', 'gxadmin', 'gxformat2', 'gxy.io', 'hack-the-galaxy-gcc2017', 'homebrew-tap', 'homebrew-toolshed', 'idc', 'infrastructure-playbook', 'it-higlass-docker', 'itcr', 'iwc', 'jxtx', 'kc-align', 'msa', 'new_user_welcome', 'nosehtml', 'p4', 'planemo', 'planemo-ci-action', 'planemo-machine', 'planemo-monitor', 'ptdk', 'pulsar', 'pulsar-helm', 'react-msa-viewer', 'readthedocs-redirect', 'SARS-CoV-2', 'screencasts', 'seek-bzip2', 'selenium-screenshots', 'sequence_utils', 'starforge', 'starforge-recipes', 'tiaas2', 'TIS_methods_review', 'tools-devteam', 'tools-iuc', 'total-perspective-vortex', 'tpv-shared-database', 'training-material', 'usage-metering', 'usegalaxy-playbook', 'usegalaxy-tools', 'VGPWorkflows', 'video-lectures', 'wheelforge']\n",
    "for repo in repositories:\n",
    "    issues = get_all_repository_issues(owner, repo)\n",
    "    if issues:\n",
    "        all_issues.extend(issues)\n",
    "\n",
    "# Convert Data to DataFrame\n",
    "df = pd.DataFrame(all_issues)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# Save DataFrame to CSV\n",
    "file_name = 'github_issues_demo_3' + owner + '.csv'\n",
    "\n",
    "df.to_csv(file_name, index=False)\n",
    "\n",
    "print(f'GitHub issues data for {owner} (open and closed) saved to github_issues.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca2d00f9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
